test/functional/wallet_conflicts.py

```diff
@@ -123,5 +130,165 @@ def run_test(self):
         assert_equal(former_conflicted["confirmations"], 1)
         assert_equal(former_conflicted["blockheight"], 217)
 
+    def test_mempool_conflict(self):
+        self.nodes[0].createwallet("alice")
+        alice = self.nodes[0].get_wallet_rpc("alice")
+
+        bob = self.nodes[1]
+
+        self.nodes[2].send(outputs=[{alice.getnewaddress() : 25} for _ in range(3)])
+        self.generate(self.nodes[2], 1)
+
+        self.log.info("Test a scenario where a transaction has a mempool conflict")
+
+        unspents = alice.listunspent()
+        assert_equal(len(unspents), 3)
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[0], unspents[1]], outputs=[{bob.getnewaddress() : 49.9999}])
+        tx1 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[1], unspents[2]], outputs=[{bob.getnewaddress() : 49.99}])
+        tx2 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        raw_tx = alice.createrawtransaction(inputs=[unspents[2]], outputs=[{bob.getnewaddress() : 24.9899}])
+        tx3 = alice.signrawtransactionwithwallet(raw_tx)['hex']
+
+        tx1_txid = alice.sendrawtransaction(tx1)
+
+        assert_equal(alice.listunspent()[0]["txid"], unspents[2]["txid"])
+        assert_equal(alice.getbalance(), 25)
+
+        tx2_txid = alice.sendrawtransaction(tx2)
+
+        # Check that the 0th unspent is now available because the transaction spending it has been replaced in the mempool
+        assert_equal(alice.listunspent()[0]["txid"], unspents[0]["txid"])
+        assert_equal(alice.getbalance(), 25)
+
+        assert_equal(alice.gettransaction(tx1_txid)["mempoolconflicts"], [tx2_txid])
+
+        self.log.info("Test scenario where a mempool conflict is removed")
+
+        alice.sendrawtransaction(tx3)
+
+        assert_equal(alice.gettransaction(tx1_txid)["mempoolconflicts"], [])
```

Comment:
71059cf4267d80d2934b85a7046fc7b64900378b

We don't expect to see tx1 in mempool, right? It's not being rebroadcast, as it's inactive?

```suggestion
        assert tx1_txid not in self.nodes[0].getrawmempool()
```
Might be good to have a check for this behavior.

Reply:
I've added an assertion that the tx isn't in the mempool.